eng/native/build-commons.sh

```diff
@@ -39,6 +39,9 @@ check_prereqs()
             # We try again with the PKG_CONFIG_PATH in place, if pkg-config still can't find OpenSSL, exit with an error, cmake won't find OpenSSL either
             pkg-config openssl || { echo >&2 "Please install openssl before running this script, see https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/macos-requirements.md"; exit 1; }
         fi
+    elif [[ "$__HostOS" == "linux" ]]; then
+        # Check presence of cmake on the path
+        command -v cmake 2>/dev/null || { echo >&2 "Please install cmake before running this script, see https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md"; exit 1; }
```

Comment:
We are already testing it in gen-buildsys.sh. Also why is this linux only?

Reply:
This check is happening earlier and is clearer. Ideally I'd like this check to happen even earlier, but I think that belongs in another PR.

Reply:
We used to had an upfront check for a while and it was deleted in https://github.com/dotnet/runtime/pull/39044.

I think if we are bringing it back, we should do it for all platforms, not just linux. CMake is needed on all platforms alike.

Reply:
We used to have an upfront check for a while and it was deleted in https://github.com/dotnet/runtime/pull/39044.

I think if we are bringing it back, we should do it for all platforms, not just linux. CMake is needed on all platforms alike.

Reply:
And this is linux-only because I think checking for pkg-config is better on Mac. What we're looking for is the thing least-likely to have been installed by-default.

Reply:
If we use `elif` vs. `else`, that will cover mac+linux+{everything-else}:

```sh
else
  # Check presence of cmake on the path
  command -v cmake 2>/dev/null || { echo >&2 "Please install cmake before running this script, see https://github.com/dotnet/runtime/tree/main/docs/workflow/requirements"; exit 1; }
```

User can then read `{platform}-instructions.md`.

Reply:
>  This check is happening earlier and is clearer.

It is rendering this check redundant which can be deleted: https://github.com/dotnet/runtime/blob/ed339b3670f6d3d971a5b56a6572d0b07095706d/eng/native/gen-buildsys.sh#L87-L88

Reply:
> We used to have an upfront check for a while and it was deleted in #39044.

Not quite the same -- that's a version check, here I'm just doing a basic check to see if it exists at all. The intent is to catch people who have not installed the prereqs at all, not people who have misconfigured prereqs.